### ISSUE TITLE
Using Float64 by default for experiments and tests

### ISSUE THREAD

**Issue by smarras79** [CliMA/ClimateMachine.jl#1589]
# Description

By some recent experiments with BOMEX carried by @slifer50, it arose that Float32 may create convergence issues in moist thermo (e.g. lack of convergence).

I would suggest that all of the default tests and experiments in master should use Float64 by default and use Float32 only for individual experiments defined by the user.

## Additional context

Add any other reasons why this should be addressed.

# For CLIMA Developers

- [ ] This feature can be added (if it cannot be, explain why in a comment below -- lack of technical expertise, not relevant to the scope of this project, too ambitious)
- [ ] There is a timeline for when this feature can be implemented
- [ ] The feature has been (or will be) implemented (Please link the PR)

**Comment by tapios** [comment]
I suspect that, as often, the convergence issues in the thermodynamics are merely a symptom of numerical instabilities, not the cause of it. It would be good to figure out why increasing the domain size leads to stability problems with single precision. Dealing with this may not require global Float64, perhaps just in some places.

**Comment by smarras79** [comment]
I agree with that and we need to work on the underlying issues; but it is still best to provide a potential user with a precision setup that will not trigger the code to break and for him to not know why.




-----------------------------------------------------------
Simone Marras, Ph.D.
Assistant Professor, 
Mechanical and Industrial Engineering
CAMS Center for Applied Mathematics and Statistics
New Jersey Institute of Technology (NJIT)
323 Dr Martin Luther King Jr Blvd, Newark, NJ 07102-1982
(Office) 973-596-3418 (MEC 315)

"Everyone here is smart, distinguish yourself by being kind”

https://scholar.google.com/citations?user=eqkE-s0AAAAJ&hl=en
https://www.researchgate.net/profile/Simone_Marras
https://github.com/smarras79/
https://njit.webex.com/meet/smarrasnjit.edu

**Comment by szy21** [comment]
I think it's fine to set the default to Float64. Having said that, @slifer50 is there any update on where the issue is? It would be nice to be able to run the simulations with Float32.

I run some simulations recently and here is the result. I'm mostly concerned about the difference in var_w above the cloud - is it acceptable? (Don't worry about the strange thl at the top or site 17 for now - that's another issue.)
![image](https://user-images.githubusercontent.com/11598433/95364618-e7596480-0885-11eb-855d-22e4bc8df011.png)
![image](https://user-images.githubusercontent.com/11598433/95364677-f9d39e00-0885-11eb-9ff6-fc7bfab3cc66.png)

**Comment by smarras79** [comment]
Good comparison to see, @szy21. The small deviation is not concerning if we consider that the Richardson correction above the cloud (which I assume is a stable atmosphere) depends on solely on the correct calculation of the vertical derivative of theta, which is sensitive to precision. 

Do you have a sponge starting at z=3.5 km?

**Comment by szy21** [comment]
> Do you have a sponge starting at z=3.5 km?
 Yes, I have a sponge starting at z=3.4km.
